visualize-admin/visualization-tool
# feat: Add Animation for Column, Pie and Scatter charts

Closes #834.

## Goals / Scope

This PR introduces Animation field which is possible to use with Column, Pie and Scatter charts (more charts to be implemented in the future) and removes `timeSlider` from interactive filters.

## Description

Additionally, I included a proposed change to use D3.js to animate columns in non-grouped and non-stacked column charts to greatly improve the interactions with the animation. I am curious what you think about this approach @ptbrowne, if you think it's fine, then I can expand this to other charts (and extract the logic to be reusable).

## Comments

I've extracted the comments from below discussion into two new issues to be tackled separately from this PR (#1029 #1030).

I think it would make sense to have some sort of UI feedback when starting stepped animation (I've thought about a progress circle around the play button that would do one lap in between the animation jumps). But I think it could also be tackled in another PR, possibly with some design support :)

## How to test

1. Go to [link](https://visualization-tool-git-feat-time-slider-continuation-2-ixt1.vercel.app/en/create/new?cube=https://environment.ld.admin.ch/foen/ubd0104/6).
2. Change horizontal axis to `Parameter`.
3. Select `Date` and Animation encoding.
4. See that it's possible to run the animation in two modes, change the duration and hide the play button.
